src/Backups/BackupCoordinationStageSync.cpp

```diff
@@ -121,8 +121,7 @@ BackupCoordinationStageSync::BackupCoordinationStageSync(
 
     try
     {
-        concurrency_check.emplace(is_restore, /* on_cluster = */ true, zookeeper_path, allow_concurrency, concurrency_counters_);
-        createStartAndAliveNodes();
+        createStartAndAliveNodesAndCheckConcurrency(concurrency_counters_);
```

Comment:
The concurrency check was improved a bit, that must help `test_backup_restore_on_cluster/test_disallow_concurrency.py::test_concurrent_backups_on_different_nodes` to stop being flaky.